### PR TITLE
Bump Kani version to 0.43.0

### PR DESCRIPTION
**Pull request by feliperodri**
## What's Changed
* Rust toolchain upgraded to `nightly-2023-12-14` by @tautschnig and @adpaco-aws

**Full Changelog**: https://github.com/model-checking/kani/compare/kani-0.42.0...kani-0.43.0

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,20 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.43.0]
+
+## What's Changed
+* Make `codegen_span` stable by @ouz-a in https://github.com/model-checking/kani/pull/2882
```
Comment (celinval) [inline CHANGELOG.md:10]: Unfortunately I don't think any of these changes are really customer visible. If you want, I would just merge all of these into one and list the contributors

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,20 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
...
+* Change reachability module to use StableMIR by @celinval in https://github.com/model-checking/kani/pull/2894
+* Fix StableMIR migration documentation by @celinval in https://github.com/model-checking/kani/pull/2915
+* Stabilize `find_debug_info` by @ouz-a in https://github.com/model-checking/kani/pull/2912
+* Migrate hooks and place modules to use mostly StableMIR APIs by @celinval in https://github.com/model-checking/kani/pull/2910
+* Migrate static handling and most of the operand codegen code to StableMIR by @celinval in https://github.com/model-checking/kani/pull/2931
+* Migrate `rvalue` and `coercion` module to use StableMIR by @celinval in https://github.com/model-checking/kani/pull/2938
+* Migrate `intrinsics` module to use StableMIR by @celinval in https://github.com/model-checking/kani/pull/2939
+
```
Comment (celinval) [inline CHANGELOG.md:18]: Missing the toolchain update

File: CHANGELOG.md (hunk 1 above)
Comment (feliperodri) [reply CHANGELOG.md:10]: I just removed all of them.

File: CHANGELOG.md (hunk 2 above)
Comment (feliperodri) [reply CHANGELOG.md:18]: I added the non-automatic toolchain updates.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,14 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.43.0]
+
+## What's Changed
+* Update the rust toolchain to nightly-2023-12-03 by @tautschnig in https://github.com/model-checking/kani/pull/2913
```
Comment (celinval) [inline CHANGELOG.md:10]: Sorry, we merge all of the toolchain upgrades into one line and list everyone that contributed to them. Please see previous releases for an example.

Also, in that line we list the current toolchain version.

Comment (feliperodri) [reply CHANGELOG.md]: Updated as suggested.

**Review by jaisnan — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,13 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.43.0]
+
+## What's Changed
+* Update the rust toolchain to `nightly-2023-12-14` by @tautschnig and @adpaco-aws
```
Comment (adpaco-aws) [inline CHANGELOG.md:10]:
```suggestion
### What's Changed
* Rust toolchain upgraded to `nightly-2023-12-14` by @tautschnig and @adpaco-aws
```

File: tests/perf/s2n-quic
Comment (adpaco-aws) [inline tests/perf/s2n-quic:1]: Is this change intended?

File: CHANGELOG.md (hunk 4 above)
Comment (feliperodri) [reply CHANGELOG.md]: Implemented as suggested.

File: tests/perf/s2n-quic
Comment (feliperodri) [reply tests/perf/s2n-quic:1]: No! Removed it.

File: CHANGELOG.md (hunk 4 above)
Comment (adpaco-aws) [reply CHANGELOG.md]: No, there should be another `#` in `## What's Changed`.

Comment (feliperodri) [reply CHANGELOG.md]: Got it!

**Review by adpaco-aws — APPROVED** [review]
